File: graal-nodejs/test/graal/unit/null.js
/*
 * Copyright (c) 2015, Oracle and/or its affiliates. All rights reserved.
 * ORACLE PROPRIETARY/CONFIDENTIAL. Use is subject to license terms.
 */

var assert = require('assert');
var module = require('./_unit');

describe('Null', function () {
    describe('IsUndefined', function () {
        it('should return false', function () {
            assert.strictEqual(module.Null_IsUndefined(), false);
        });
    });
    describe('IsNull', function () {
        it('should return true', function () {
            assert.strictEqual(module.Null_IsNull(), true);
        });
    });
    describe('IsTrue', function () {
        it('should return false', function () {
            assert.strictEqual(module.Null_IsTrue(), false);
        });
    });
    describe('IsFalse', function () {
        it('should return false', function () {
            assert.strictEqual(module.Null_IsFalse(), false);
        });
    });
    describe('IsNativeError', function () {
        it('should return false', function () {
            assert.strictEqual(module.Value_IsNativeError(null), false);
        });
    });
});

File: graal-nodejs/test/graal/unit/null.cc
/*
 * Copyright (c) 2015, Oracle and/or its affiliates. All rights reserved.
 * ORACLE PROPRIETARY/CONFIDENTIAL. Use is subject to license terms.
 */

#define SUITE Null

// Null/MissingPrimitive::IsUndefined

EXPORT_TO_JS(IsUndefined) {
    args.GetReturnValue().Set(Null(args.GetIsolate())->IsUndefined());
}

// Null/MissingPrimitive::IsNull

EXPORT_TO_JS(IsNull) {
    args.GetReturnValue().Set(Null(args.GetIsolate())->IsNull());
}

// Null/MissingPrimitive::IsTrue

EXPORT_TO_JS(IsTrue) {
    args.GetReturnValue().Set(Null(args.GetIsolate())->IsTrue());
}

// Null/MissingPrimitive::IsFalse

EXPORT_TO_JS(IsFalse) {
    args.GetReturnValue().Set(Null(args.GetIsolate())->IsFalse());
}

#undef SUITE
